Source/NameServer/AddressBook/AddressBook_Record.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef AddressBook_Record_H
+#define AddressBook_Record_H
+
+#include "AddressBook_Defs.hpp"
+
+namespace AddressBook
+{
+enum RecordType_t { Device = 0, DeviceExt, External, Supervisor };
+
+// data contents factored out for serialisation 17 July 2019 ADR  
+struct RecordData_t
+{
+   public:
+ 
+   SymAddr_t Address;          // Device's full sym address.
+   union                       // Disambiguated by RecordType, which appears at
+   {                           // at the end of the struct for packing reasons.
+       SymAddr_t Supervisor;    // Full sym address of the Device's Supervisor.
+       unsigned long Rank;	    // OR Supervisor's MPI Rank. 
+   };                          
+   DTypeIdx DeviceType;        // Index of the Device's type in task.
+   AttrIdx Attribute;          // Index of the Device's attribute in task.
```

Comment:
I don't understand this comment - what is a "Device's attribute"?

Reply:
A device can supposedly have a free-form attribute that can be stored - this facilitates that

Reply:
What does that mean in the wider context of POETS?